src/overlays/actors/ovl_En_Scopecrow/z_en_scopecrow.c

```diff
@@ -31,47 +30,324 @@ const ActorInit En_Scopecrow_InitVars = {
+}
+
+s32 func_80BCD09C(s16 arg0) {
+    switch (arg0) {
+        case 0:
+            if (gSaveContext.save.weekEventReg[53] & 4) {
+                return true;
+            }
+            return false;
+
+        case 1:
+            if (gSaveContext.save.weekEventReg[53] & 0x80) {
+                return true;
+            }
+            return false;
+
+        case 2:
+            if (gSaveContext.save.weekEventReg[54] & 1) {
+                return true;
+            }
+            return false;
+
+        case 3:
+            if (gSaveContext.save.weekEventReg[54] & 2) {
+                return true;
+            }
+            return false;
+
+        case 4:
+            if (gSaveContext.save.weekEventReg[54] & 4) {
+                return true;
+            }
+            return false;
+
+        case 5:
+            if (gSaveContext.save.weekEventReg[54] & 8) {
+                return true;
+            }
+            return false;
+    }
+
+    return false;
+}
+
+s32 func_80BCD1AC(s16 arg0) {
+    switch (arg0) {
+        case 0:
+            if (!(gSaveContext.save.weekEventReg[53] & 4)) {
+                gSaveContext.save.weekEventReg[53] |= 4;
+                return true;
+            }
+            break;
+
+        case 1:
+            if (!(gSaveContext.save.weekEventReg[53] & 0x80)) {
+                gSaveContext.save.weekEventReg[53] |= 0x80;
+                return true;
+            }
+            break;
+
+        case 2:
+            if (!(gSaveContext.save.weekEventReg[54] & 1)) {
+                gSaveContext.save.weekEventReg[54] |= 1;
+                return true;
+            }
+            break;
+
+        case 3:
+            if (!(gSaveContext.save.weekEventReg[54] & 2)) {
+                gSaveContext.save.weekEventReg[54] |= 2;
+                return true;
+            }
+            break;
+
+        case 4:
+            if (!(gSaveContext.save.weekEventReg[54] & 4)) {
+                gSaveContext.save.weekEventReg[54] |= 4;
+                return true;
+            }
+            break;
+
+        case 5:
+            if (!(gSaveContext.save.weekEventReg[54] & 8)) {
+                gSaveContext.save.weekEventReg[54] |= 8;
+                return true;
+            }
+            break;
+    }
+
+    return false;
+}
+
+void func_80BCD2BC(EnScopecrow* this, GlobalContext* globalCtx) {
+    Actor_SpawnAsChildAndCutscene(&globalCtx->actorCtx, globalCtx, ACTOR_EN_SC_RUPPE, this->actor.world.pos.x,
+                                  this->actor.world.pos.y, this->actor.world.pos.z, this->actor.shape.rot.x,
+                                  this->actor.shape.rot.y, this->actor.shape.rot.z, this->actor.params,
+                                  this->actor.cutscene, this->actor.unk20, NULL);
+}
+
+s32 func_80BCD334(EnScopecrow* this, Path* path, s32 count_) {
+    Vec3s* points = Lib_SegmentedToVirtual(path->points);
+    s32 sp58 = path->count;
+    s32 count = count_;
```

Comment:
Looks like `sp58` should be the count, not arg2. For these functions I usually just call the arg2 `pointIndex` and the stack variable `index`.